From a5ee18c5d26e848ceca454d15b742260ef27dab0 Mon Sep 17 00:00:00 2001
From: Scott Weiss <sdw35@cornell.edu>
Date: Wed, 20 Sep 2017 13:59:56 -0400
Subject: [PATCH] use forked versions for manageiq-providers-{os,k8s},
 ui-classic

---
 Gemfile | 12 +++++++++---
 1 file changed, 9 insertions(+), 3 deletions(-)

diff --git a/Gemfile b/Gemfile
index 9028e563e6..3b1022b145 100644
--- a/Gemfile
+++ b/Gemfile
@@ -17,6 +17,12 @@ def manageiq_plugin(plugin_name)
   end
 end
 
+def cm_integration_plugin(plugin_name)
+  unless dependencies.detect { |d| d.name == plugin_name }
+    gem plugin_name, :git => "https://github.com/container-mgmt/#{plugin_name}", :branch => "image-unstable"
+  end
+end
+
 manageiq_plugin "manageiq-providers-ansible_tower" # can't move this down yet, because we can't autoload ManageIQ::Providers::AnsibleTower::Shared
 manageiq_plugin "manageiq-schema"
 
@@ -105,7 +111,7 @@ group :hawkular, :manageiq_default do
 end
 
 group :kubernetes, :openshift, :manageiq_default do
-  manageiq_plugin "manageiq-providers-kubernetes"
+  cm_integration_plugin "manageiq-providers-kubernetes"
 end
 
 group :lenovo, :manageiq_default do
@@ -121,7 +127,7 @@ group :qpid_proton, :optional => true do
 end
 
 group :openshift, :manageiq_default do
-  manageiq_plugin "manageiq-providers-openshift"
+  cm_integration_plugin "manageiq-providers-openshift"
   gem "htauth",                         "2.0.0",         :require => false # used by container deployment
 end
 
@@ -184,7 +190,7 @@ group :consumption, :manageiq_default do
 end
 
 group :ui_dependencies do # Added to Bundler.require in config/application.rb
-  manageiq_plugin "manageiq-ui-classic"
+  cm_integration_plugin "manageiq-ui-classic"
   # Modified gems (forked on Github)
   gem "jquery-rjs",                   "=0.1.1",                       :git => "https://github.com/ManageIQ/jquery-rjs.git", :tag => "v0.1.1-1"
 end
-- 
2.14.3

